src/App.tsx

```diff
@@ -1,16 +1,54 @@
+// import { AboutMe } from './components/AboutMe/AboutMe.tsx';
+import { useState } from 'react';
+
 import { AboutMe } from './components/AboutMe/AboutMe.tsx';
 import { Footer } from './components/Footer/Footer.tsx';
 import { HeaderComponent } from './components/HeaderComponent/HeaderComponent.tsx';
+import { ProductsList } from './components/ProductsList/ProductsList.tsx';
+import { PRODUCTS_DATA } from './data/data.ts';
+import type { Product } from './interfaces/Product.ts';
 
 // import styles from './App.module.css';
 
+export type Pages = 'about' | 'products';
+
 function App() {
+    const localCart = localStorage.getItem('cart') ? JSON.parse(localStorage.getItem('cart')!) : [];
+
+    const [currentPage, setCurrentPage] = useState<Pages>('about');
+    const [cartData, setCartData] = useState<Array<Product> | []>(localCart);
```

Comment:
it will be readable, do not add empty array `[]` as a type, never
```suggestion
    const [cartData, setCartData] = useState<Product[]>(localCart);
```